pkg/graph/resolvers/plan_document.go

```diff
@@ -0,0 +1,80 @@
+package resolvers
+
+import (
+	"github.com/cmsgov/mint-app/pkg/graph/model"
+	"github.com/cmsgov/mint-app/pkg/models"
+	"github.com/cmsgov/mint-app/pkg/storage"
+	"github.com/cmsgov/mint-app/pkg/storage/genericmodel"
+	"github.com/cmsgov/mint-app/pkg/upload"
+	"github.com/google/uuid"
+	"go.uber.org/zap"
+)
+
+func createDocumentPayload(s3Client *upload.S3Client, document *models.PlanDocument) (*model.PlanDocumentPayload, error) {
+	presignedURL, urlErr := s3Client.NewGetPresignedURL(*document.FileKey)
+	if urlErr != nil {
+		return nil, urlErr
+	}
+
+	payload := model.PlanDocumentPayload{
+		Document:     document,
+		PresignedURL: presignedURL,
+	}
+
+	return &payload, nil
+}
+
+// PlanDocumentCreate implements resolver logic to create a plan document object
+func PlanDocumentCreate(logger *zap.Logger, input *model.PlanDocumentInput, principal *string, store *storage.Store, s3Client *upload.S3Client) (*model.PlanDocumentPayload, error) {
```

Comment:
I think for consistency, this shouldn't take the generated `model.PlanDocumentInput`, but instead take the `models.PlanDocument`. The other resolvers convert the input to the hand rolled object in schema.resolvers.go.